src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -304,6 +308,15 @@ public override Task<T> WaitForExternalEvent<T>(string name)
                     reason: $"WaitForExternalEvent:{name}",
                     isReplay: this.innerContext.IsReplaying);
 
+                // Check the queue to see if any events came in before the orchestrator was listening
+                if (this.bufferedExternalEvents.TryGetValue(name, out Queue queue))
+                {
+                    object input = queue.Dequeue();
+
+                    // We can call raise event right away, since we already have an event's input
+                    this.RaiseEvent(name, input.ToString());

```

Comment:
If we're able to raise the event right away, then we probably don't need to trace the `FunctionListening` event above.